cmake/modules/hunter_calculate_config_sha1.cmake

```diff
@@ -20,57 +20,52 @@ function(hunter_calculate_config_sha1 hunter_self hunter_base user_config)
 
   hunter_status_print("Calculating Config-SHA1")
 
-  # Include default_config
-  set(default_config "${hunter_self}/cmake/configs/default.cmake")
-  if(NOT EXISTS "${default_config}")
-    hunter_internal_error("File `${default_config}` not exists")
-  endif()
-  set(HUNTER_ALLOW_CONFIG_LOADING YES)
-  include("${default_config}")
-  set(HUNTER_ALLOW_CONFIG_LOADING NO)
+  # Include config from alternate repos
+  foreach(repo "${hunter_self}/cmake" ${HUNTER_RECIPE_DIRS})
+    if(EXISTS "${repo}/configs/default.cmake")
+      set(configname "${repo}/configs/default.cmake")
+    else()
+      hunter_internal_error("Could not find default.cmake for ${repo}")
+    endif()
+    set(HUNTER_ALLOW_CONFIG_LOADING YES)
+    include("${configname}")
+    set(HUNTER_ALLOW_CONFIG_LOADING NO)
+  endforeach()
 
-  # Include user_config
+  # Include user_config: this is a filename, so it doesn't fit in the pattern
+  # above
   if(NOT EXISTS "${user_config}")
-    hunter_internal_error("Hunter config not exists")
+    hunter_internal_error("Hunter config does not exist")
   endif()
   set(HUNTER_ALLOW_CONFIG_LOADING YES)
   include("${user_config}")
   set(HUNTER_ALLOW_CONFIG_LOADING NO)
 
   # Create list of the projects
-  set(directory_with_projects "${hunter_self}/cmake/projects")
-  if(NOT EXISTS "${directory_with_projects}")
-    hunter_internal_error("Directory `${directory_with_projects}` not exists")
-  endif()
-  if(NOT IS_DIRECTORY "${directory_with_projects}")
-    hunter_internal_error(
-        "Path `${directory_with_projects}` is not a directory"
-    )
-  endif()
-
-  file(
-      GLOB projects
-      RELATIVE "${directory_with_projects}"
-      "${directory_with_projects}/*"
-  )
-
-  if(NOT projects)
-    hunter_internal_error("No projects found")
-  endif()
-
-  set(real_projects "")
-  foreach(x ${projects})
-    if(IS_DIRECTORY "${directory_with_projects}/${x}")
-      list(APPEND real_projects "${x}")
+  foreach(repo ${HUNTER_RECIPE_DIRS} "${hunter_self}/cmake")
+    if(NOT EXISTS "${repo}/projects")
+      hunter_internal_error("Directory `${repo}/projects` not exists")
+    endif()
+    if(NOT IS_DIRECTORY "${repo}/projects")
+      hunter_internal_error("Path `${repo}/projects` is not a directory")
     endif()
+    file(GLOB subdirectories RELATIVE "${repo}/projects" "${repo}/projects/*/")
```

Comment:
It would an archive of packages, separate from those in Hunter. It mainly meant to reference packages that are not currently in Hunter, whether because those packages are private, or because they have not met their way there (think specialised community). Once repos are added via the syntax above, the glob is as much of an issue as in the current code. The rebuild happens because any change to a repo means a change to its hash, as with HunterID.